src/ui/public/directives/input_base64.js

```diff
@@ -0,0 +1,88 @@
+import { uiModules } from 'ui/modules';
+import { Observable } from 'rxjs/Rx';
+const module = uiModules.get('kibana');
+
+const createFileContent$ = (file) => {
+  return Observable.create(observer => {
+    const reader = new FileReader();
+    reader.onerror = (err) => {
+      observer.error(err);
+    };
+
+    reader.onload = () => {
+      observer.next(reader.result);
+      observer.complete();
+    };
+
+    reader.readAsDataURL(file);
+
+    return () => {
+      reader.abort();
+    };
+  });
+};
+
+module.directive('inputBaseSixtyFour', function () {
+  return {
+    restrict: 'A',
+    require: 'ngModel',
+    scope: 'isolate',
+    link: function ($scope, $elem, attrs, ngModel) {
+      const maxSizeValidator = (dataUrl) => {
+        const base64ToKb = (length) => {
+          const totalBits = length * 6; // 6 bits per character in base64
+          return totalBits / 8 / 1024;
+        };
+
+        return {
+          errorKey: 'maxSize',
+          isValid: base64ToKb(dataUrl.length) <= attrs.max
+        };
+      };
+
+      const validators = [ maxSizeValidator ];
+
+      // produce fileContent$ whenever the $element 'change' event is triggered.
+      const fileContent$ = Observable
+        .fromEvent($elem, 'change')
+        .map(e => e.target.files)
+        .switchMap(files => {
+          if (files.length === 0) {
+            return [];
+          }
+
+          if (files.length > 1) {
+            throw new Error('Only one file is supported at this time');
```

Comment:
Couldn't this be a validator rather than a fatal error?

Reply:
Also, is it possible to submit multiple files since the `input[type=file]` requires the `multiple` attribute to enable multi-selection?

Reply:
I don't really want this error to be displayed to the end-user as much as the developer who tries to use it on an `input` with the `multiple` attribute, so I don't think it makes much sense to have it as a validator.

Reply:
We could, potentially, but it makes the code more complicated to support multiple files (when we don't really need it yet) so I'm throwing "better" errors more places to make this more obvious.

Reply:
👍